src/js/components/modals/ServiceScaleFormModal.js

```diff
@@ -106,6 +106,17 @@ class ServiceScaleFormModal extends mixin(StoreMixin) {
           id: service.id,
           scaleBy: parseInt(instances, 10)
         });
+      } else if (service instanceof Pod) {
+        let spec = Object.assign({}, service.getSpec().get(),
```

Comment:
I introduced `MarathonSpecUtil` for all the spec-related mutations. Let me know if that follows your thinking.